File: business/portal/backend/models/column.go
package models

import (
	"encoding/json"

	"gorm.io/gorm"
)

type Column struct {
	gorm.Model
	Name        string   `gorm:"size:100;not null" json:"name"`
	Code        string   `gorm:"size:100;not null" json:"code"`
	PageID      uint     `gorm:"not null;index" json:"pageId"`
	ParentID    uint     `gorm:"default:0;index" json:"parentId"`
	RoutePath   string   `gorm:"size:200" json:"routePath"`
	Template    string   `gorm:"size:100" json:"template"`
	Description string   `gorm:"size:500" json:"description"`
	Sort        int      `gorm:"default:0" json:"sort"`
	Status      int      `gorm:"default:1" json:"status"`
	DisplayType int      `gorm:"default:1" json:"displayType"`
	WorkflowID  *uint    `gorm:"index" json:"workflowId"`
	Workflow    Workflow `gorm:"foreignKey:WorkflowID;references:ID" json:"workflow"`
}

func (c Column) MarshalJSON() ([]byte, error) {
	type Alias Column
	return json.Marshal(&struct {
		CreatedAt string `json:"createTime"`
		UpdatedAt string `json:"updatedAt"`
		Alias
	}{
		CreatedAt: c.CreatedAt.Format("2006-01-02 15:04:05"),
		UpdatedAt: c.UpdatedAt.Format("2006-01-02 15:04:05"),
		Alias:     (Alias)(c),
	})
}

File: business/portal/frontend/src/api/column.ts
import request from '@/utils/request'

export interface ColumnForm {
  id?: number
  name: string
  code: string
  pageId: number
  parentId?: number
  routePath?: string
  template?: string
  description?: string
  sort: number
  status: number
  displayType: number
  workflowId?: number
}

export function getColumns(params?: { pageId?: number; parentId?: number; displayType?: number }) {
  return request.get('/columns', { params })
}

export function getColumnPublishes() {
  return request.get('/columns/publishes')
}

export function createColumn(data: ColumnForm) {
  return request.post('/columns', data)
}

export function updateColumn(id: number, data: ColumnForm) {
  return request.put(`/columns/${id}`, data)
}

export function deleteColumn(id: number) {
  return request.delete(`/columns/${id}`)
}
